src/sinks/file/file.rs

```diff
@@ -2,37 +2,48 @@ use super::Encoding;
 use crate::event::{self, Event};
 use bytes::{Bytes, BytesMut};
 use codec::BytesDelimitedCodec;
-use futures::{try_ready, Async, AsyncSink, Future, Poll, Sink, StartSend};
-use std::{ffi, io, path};
-use tokio::{codec::Encoder, fs::file, io::AsyncWrite};
+use futures::{future, try_ready, Async, AsyncSink, Future, Poll, Sink, StartSend};
+use std::{ffi, fmt, io, path};
+use tokio::{
+    codec::Encoder,
+    fs::{self, file},
+    io::AsyncWrite,
+};
 
 const INITIAL_CAPACITY: usize = 8 * 1024;
 const BACKPRESSURE_BOUNDARY: usize = INITIAL_CAPACITY;
 
-#[derive(Debug)]
 pub struct File {
     state: State,
     encoding: Encoding,
     buffer: BytesMut,
     codec: BytesDelimitedCodec,
 }
 
-#[derive(Debug)]
 enum State {
-    Creating(file::OpenFuture<BytesPath>),
+    Creating(Box<dyn Future<Item = file::File, Error = io::Error> + Send + 'static>),
     Open(file::File),
     Closed,
 }
 
 impl File {
     pub fn new(path: Bytes, encoding: Encoding) -> Self {
         let path = BytesPath(path);
+        let parent = path.as_ref().parent().map(|p| p.to_path_buf());
 
-        let fut = file::OpenOptions::new()
+        let dir_fut = if let Some(parent) = parent {
+            future::Either::A(fs::create_dir_all(parent))
```

Comment:
Will this do the create even when the directory already exists? Probably not a problem, just want to be clear.

Reply:
What was the resolution here?

Reply:
Oops sorry I misread this yesterday, I believe it acts like `mkdir -p` which will not return an error if the directory already exists.

Each test calls this when the directory exists and when it doesn't so its safe to run if the dir exists or not.